src/controller/python/BUILD.gn

```diff
@@ -226,6 +226,10 @@ chip_python_wheel_action("chip-core") {
         "chip/storage/__init__.py",
         "chip/utils/CommissioningBuildingBlocks.py",
         "chip/utils/__init__.py",
+        "chip/yaml/DataModelLookup.py",
+        "chip/yaml/YamlParser.py",
+        "chip/yaml/YamlUtils.py",
+        "chip/yaml/__init__.py",
```

Comment:
Please coordinate with @vivien-apple in terms of the requirements we have for running YAML bits, including in settings where the client cannot run Python itself, but could be talked to by something running Python.  It would be good to be able to use the Python YAML parser in that setting.

Reply:
I will try to sync up with Vivien on that today. 

Overall I think this is very much doable. I can start restructuring a little in a follow up PR, but overall all I think we need to do is separate the execution and validation, which giving the ability to switch out the "execution" part. For python we continue doing what we are doing, for the section that needs to talk to an external chip tool it can do so in whatever way Vivien was already planning on doing there.

Reply:
> including in settings where the client cannot run Python itself

Are there any details about such settings?

Reply:
Resolving this comment - while we should move towards something that is usable generically, running YAML tests from python is by itself very useful as a stand alone functionality, so PR should not be blocked on this.